src/polyfill.web.tsx

```diff
@@ -1 +1,56 @@
-export * from './mock';
\ No newline at end of file
+import * as React from 'react';
+import { View } from 'react-native';
+import { AppearancePreferences, ColorSchemeName, AppearanceListener } from './Appearance.types';
+
+type Mode = 'light' | 'dark' | 'no-preference';
+
+const isNoPreferenceQuery = window.matchMedia('(prefers-color-scheme: no-preference)');
+const isDarkQuery = window.matchMedia('(prefers-color-scheme: dark)');
+const isLightQuery = window.matchMedia('(prefers-color-scheme: light)');
+const isSupported = isDarkQuery.matches || isNoPreferenceQuery.matches || isLightQuery.matches;
+
+const initialMode: Mode = isSupported
+  ? window.matchMedia('(prefers-color-scheme: dark)').matches
+    ? 'dark'
+    : 'light'
+  : 'no-preference';
+
+interface EventSubscription {
+  remove: () => void;
+}
+
+function removeListener() {
+  window.matchMedia('(prefers-color-scheme: light)').removeListener(null);
+  window.matchMedia('(prefers-color-scheme: dark)').removeListener(null);
+  window.matchMedia('(prefers-color-scheme: no-preference)').removeListener(null);
+}
+
+export class Appearance {
+  static getColorScheme(): ColorSchemeName {
+    return initialMode;
+  }
+
+  static set(_preferences: AppearancePreferences): void {}
+
+  static addChangeListener(_listener: AppearanceListener): EventSubscription {
+    isDarkQuery.addListener(e => {
+      const colorScheme = e.matches ? 'dark' : 'light';
+      _listener({ colorScheme });
+      this.set({ colorScheme });
+    });
+    return { remove: () => removeListener };
+  }
+
+  /**
+   * Unused: some people might expect to remove the listener like this, but they shouldn't.
+   */
+  static removeChangeListener(_listener: AppearanceListener): void {
+    removeListener();
+  }
+}
+
+export const AppearanceProvider = (props: any) => <View style={{ flex: 1 }} {...props} />;
```

Comment:
Try using a `React.forwardRef` here so we can cut down on `div`s and avoid adding unexpected styling behavior. I can see this becoming a problem if someone is trying to enable/disable body scrolling.